### PR TITLE
App with jetpack compose UI

### PR DESCRIPTION
**Pull request by Zermelo-101**
Changing the UI framework to JetapckCompose.
The greetingActivityTest  may seem far fetch, from what I saw they are no straight forward way to use an activity launched by a Intent using Compose

### CODE REVIEW

File: app/src/main/java/com/github/emilehreich/bootcamp/GreetingActivity.kt [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_greeting)
-
-        val greet = findViewById<TextView>(R.id.greetingName)
         val name = intent.getStringExtra("name")
+        setContent{
+            Text(text = "Welcome $name")
+        }
```
Comment (yasmineben510) [inline app/src/main/java/com/github/emilehreich/bootcamp/GreetingActivity.kt:19]: How do you make sure the created text is centered/how do you set its position? Is it by default at the center of the activity?

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt [hunk 2]
```diff
@@ -21,23 +28,16 @@ import org.junit.runner.RunWith
...
     @get:Rule
-    val activityScenarioRule = ActivityScenarioRule(MainActivity::class.java)
+    val composeTestRule = createAndroidComposeRule<MainActivity>()
 
-    @Test
-    fun testWritingOfName() {
-        onView(ViewMatchers.withId(R.id.mainText)).perform(ViewActions.replaceText("Steve"))
-        closeSoftKeyboard()
-        onView(withText("Steve")).check(matches(isDisplayed()))
-    }
```
Comment (yasmineben510) [inline app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt:33]: Maybe you should add a test to verify that the written name is displayed while being written, before clicking on the next button

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/GreetingActivityTest.kt [hunk 3]
```diff
@@ -1,28 +1,55 @@
...
+    fun setUp() {
+
+        scenario = ActivityScenario.launch(
+            createActivityIntent(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+            )
+        )
+    }
+
+    private fun createActivityIntent(
```
Comment (yasmineben510) [inline app/src/androidTest/java/com/github/emilehreich/bootcamp/GreetingActivityTest.kt:40]: This seems like the correct workaround

File: app/src/main/java/com/github/emilehreich/bootcamp/GreetingActivity.kt (hunk 1 above)
Comment (Zermelo-101) [reply app/src/main/java/com/github/emilehreich/bootcamp/GreetingActivity.kt:19]: In fact I did not. I will add it in the next commit

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt (hunk 2 above)
Comment (Zermelo-101) [reply app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt:33]: I did not understand the purpose of the test. Can you elaborate ?

Comment (yasmineben510) [reply app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt:33]: Like test if when the user writes his name in the plain text component, it is indeed modified in the view and he can see the new text he just typed before submitting it by clicking on the button. You should be able to do so by performing the replacement of the Text and then checking for the isDisplayed or equivalent

Comment (Zermelo-101) [reply app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt:33]: Seems like you add the comment on the code from the previous commit that is why I did not understand. You are right. 
NB i believe  performTextReplacement method throws an error if the code is not correctly replaced.

Comment (yasmineben510) [reply app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt:33]: Ah yes my bad!
Oh okay good to know

**Pull request closed without merging**